### PR TITLE
Ubuntu 18 build support

### PR DESCRIPTION
**Pull request by rkimballn1**
Works for me ;)

### CODE REVIEW

File: cmake/external_mkldnn.cmake [hunk 1]
```diff
@@ -48,6 +48,7 @@ set(MKLVERSION "2018.0.3.20180406")
 if (${CMAKE_SYSTEM_NAME} STREQUAL "Linux")
     set(MKLPACKAGE "mklml_lnx_${MKLVERSION}.tgz")
     set(MKL_SHA1_HASH aea0d9ce65773cfcf5d8292b8db553bde965fc8f)
+    set(MKL_LIBS libiomp5.so libmklml_intel.so)
```
Comment (aprocter) [inline cmake/external_mkldnn.cmake]: I'm guessing a similar line will be needed for `Darwin` but we can deal with that later.

File: src/resource/main.cpp [hunk 2]
```diff
@@ -94,18 +94,18 @@ int main(int argc, char** argv)
 #endif
     include_paths.push_back({NGRAPH_HEADERS_PATH, {}, true});
     include_paths.push_back({CLANG_BUILTIN_HEADERS_PATH, {}, true});
-    include_paths.push_back({"/Library/Developer/CommandLineTools/usr/include/c++/v1", {}});
+// include_paths.push_back({"/Library/Developer/CommandLineTools/usr/include/c++/v1", {}});
```
Comment (aprocter) [inline src/resource/main.cpp]: Was this change intentional?

File: cmake/external_mkldnn.cmake (hunk 1 above)
Comment (rkimballn1) [reply cmake/external_mkldnn.cmake]: yep

Comment (rkimballn1) [reply cmake/external_mkldnn.cmake]: and windows too

File: src/resource/main.cpp (hunk 2 above)
Comment (rkimballn1) [reply src/resource/main.cpp]: Yes. No longer embedding system headers in the resource file, just the headers that are from the ngraph directory. The system headers still exist after you delete the repo directory but the ngraph headers don't so they must be preserved. We COULD "install" the ngraph headers and solve the problem that way.
